### PR TITLE
Remove PRE_60_NODE_CHECKPOINT

### PR DESCRIPTION
**Pull request by DaveCTurner**
This commit removes the obsolete `PRE_60_NODE_CHECKPOINT` constant for dealing
with 5.x nodes' lack of sequence number support.

Backport of #42527

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 1]
```diff
@@ -981,28 +981,19 @@ public ReplicaResponse(long localCheckpoint, long globalCheckpoint) {
...
-            } else {
-                // 5.x used to read empty responses, which don't really read anything off the stream, so just do nothing.
-                localCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
-            if (in.getVersion().onOrAfter(Version.V_6_0_0_rc1)) {
-                globalCheckpoint = in.readZLong();
-            } else {
-                globalCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
+            assert in.getVersion().onOrAfter(Version.V_6_0_0_alpha1);
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: The later (stronger) assertion should cover this assertion. I am not sure about the usefulness of these assertions.

File: server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 2]
```diff
@@ -981,28 +981,19 @@ public ReplicaResponse(long localCheckpoint, long globalCheckpoint) {
...
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
-            if (out.getVersion().onOrAfter(Version.V_6_0_0_alpha1)) {
-                out.writeZLong(localCheckpoint);
-            }
-            if (out.getVersion().onOrAfter(Version.V_6_0_0_rc1)) {
-                out.writeZLong(globalCheckpoint);
-            }
+            assert out.getVersion().onOrAfter(Version.V_6_0_0_alpha1);
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: same comment here

**Review by dnhatn — APPROVED** [review]
I left a smaller comment but LGTM. Thanks @DaveCTurner.

File: server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: Yeah I know what you mean. I want to have _something_ here for documentation and to save the heartache that arises from tracking down serialisation issues, even though in theory we should never hit these. And then I felt the need to double-check that `rc1` was definitely a tighter check. I'll drop the looser one.

File: server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 3]
```diff
@@ -981,28 +981,17 @@ public ReplicaResponse(long localCheckpoint, long globalCheckpoint) {
...
-            } else {
-                // 5.x used to read empty responses, which don't really read anything off the stream, so just do nothing.
-                localCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
-            if (in.getVersion().onOrAfter(Version.V_6_0_0_rc1)) {
-                globalCheckpoint = in.readZLong();
-            } else {
-                globalCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
+            assert in.getVersion().onOrAfter(Version.V_6_0_0_rc1);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: What's the point of this assertion? Let's drop this.

File: server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 4]
```diff
@@ -981,28 +981,17 @@ public ReplicaResponse(long localCheckpoint, long globalCheckpoint) {
...
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
-            if (out.getVersion().onOrAfter(Version.V_6_0_0_alpha1)) {
-                out.writeZLong(localCheckpoint);
-            }
-            if (out.getVersion().onOrAfter(Version.V_6_0_0_rc1)) {
-                out.writeZLong(globalCheckpoint);
-            }
+            assert out.getVersion().onOrAfter(Version.V_6_0_0_rc1);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: same here

**Review by ywelsch — APPROVED** [review]
Left one comment. If addressed, LGTM

**Pull request merged**